clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -923,6 +923,8 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
   PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
 
   ModulePassManager MPM;
+  // Add a verifier pass, before any other passes, to catch CodeGen issues.
+  MPM.addPass(VerifierPass());
```

Comment:
I agree that it's valuable to do verification right out of IR generation, but the verifier pass is pretty heavyweight, and I don't think we want to be doing it unconditionally in release builds.  There's a reason there's an option to enable/disable it.

Reply:
Yeah, you are right, it does add some non-insignificant cost: http://llvm-compile-time-tracker.com/compare.php?from=e0cb340f8ef77e2fefa42a3d6f4bd9aab42ad660&to=e728912a39421291d44f94c8304789710fe7bb67&stat=instructions:u

Though I believe these tests are all lightweight on Sema, don't use templates much, so the cost probably would not show as bad in the problematic case.

The docs encourage frontends to run the verifier before any optimizations: https://llvm.org/docs/Passes.html#verify-module-verifier

And this post https://discourse.llvm.org/t/should-clang-verify-ir-when-passing-a-llvm-ir-file/62926/3, though in a different context, does appear favorable towards the verifier, versus cost.

Reply:
There is actually just an option to disable it, it's otherwise on by default.
See this post for more details: https://discourse.llvm.org/t/rfc-running-llvm-verifier-right-out-of-codegen/73823/4

I have changed the current MR to also add the new pass under the same flag, as it's useful to allow folks to disable it, but it's otherwise going to be on for the majority of cases.